packages/apps/plugins/plugin-stack/src/props.ts

```diff
@@ -0,0 +1,33 @@
+//
+// Copyright 2023 DXOS.org
+//
+
+import { subscribe, ObservableArray } from '@dxos/observable-object';
+
+export type StackSections = ObservableArray<StackSectionModel>;
+
+export type StackSectionModel = {
+  source: { resolver: string; guid: string };
+  object: unknown;
+};
+
+export type StackModel = {
+  id: string;
+  sections: ObservableArray<StackSectionModel>;
+};
+
+export type StackProperties = {
```

Comment:
why have both properties and model? i.e., why isn't title just part of model?

Reply:
I need to double-check on this – I was following a pattern-of-precedent in MarkdownPlugin, but need to sync with @wittjosiah on the reasoning for it.

Reply:
my suspicion was this is a leakage from the way ECHO objects are structured. and that in our striving to be pure, plugins should just define THEIR required data shape and something then does the mapping. i think this is important for us to be explicit about re proving the plugin model. otherwise we're just pretending to be decoupled from ECHO?

Reply:
@thure So for the markdown plugin the reason its separate is that I had a hard time convincing myself to pollute the `ComposerModel` contract with things its not directly concerned with. Example of this is title, there's no title in the composer model because its the contract for the editor itself. That meant that extra metadata for that model like title, etc. needed to be passed in as a separate parameter.